docs/debugging-transactions.md

```diff
@@ -0,0 +1,510 @@
+      (+) add watch expression (`+:<expr>`), (-) remove watch expression (-:<expr>)
+      (?) list existing watch expressions and breakpoints
+      (v) print variables and values (`v [bui|glo|con|loc]*`)
+      (:) evaluate expression - see `v`, (h) print this help
+      ```
+5. **Fix Issues and Re-Test**:
+    - Identify and fix any issues in your smart contracts.
+    - Re-run the debugger to verify the changes and ensure the correct behavior.
+    - Continue testing and debugging until you are satisfied with the results.
+
+### Example of Debugging a Transaction
+
+Here’s a simple example of how to deploy and interact with a contract in Truffle:
+
+1. Modify the `migrations/1_deploy_contracts.js` script to deploy and interact with the `SimpleStorage` contract:
+   ```javascript
+   const SimpleStorage = artifacts.require("SimpleStorage");
+   
+   module.exports = async function (deployer) {
+      await deployer.deploy(SimpleStorage);
+      const simpleStorage = await SimpleStorage.deployed();
+      console.log("Contract Address:", simpleStorage.address);
+      const result = await simpleStorage.set(42);
+      console.log("Transaction Hash:", result.tx);
+      const storedData = await simpleStorage.get();
+      console.log("Stored Data:", storedData.toNumber());
+   };
+   ```
+2. Run the Truffle script to deploy and interact with the contract:
+   ```bash
+   truffle migrate
+   ```
+3. Debug the transaction using the Truffle debugger:
+   ```bash
+   truffle debug <place_tx_hash_here>
+   ```
+4. Use the debugger commands to step through the transaction and inspect the state changes.
+5. Example output after running the debugger:
+   ```bash
+   SimpleStorage.sol:
+   
+   3: pragma solidity ^0.8.0;
+   4:
+   5: contract SimpleStorage {
+   ^^^^^^^^^^^^^^^^^^^^^^^^
+   
+   debug(development:0xe122240e...)> # Clicking enter here will step to the next line
+   ```
+6. Output after clicking enter several times to step through the transaction:
+   ```bash
+   SimpleStorage.sol:
+   
+   6:     uint256 public storedData;
+   7:
+   8:     function set(uint256 x) public {
+   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   
+   debug(development:0xe122240e...)>
+   
+   Transaction completed successfully.
+   ```
+7. Done! You have successfully debugged a transaction using the Truffle debugger.
+
+## 3. Advanced Debugging with `debug_traceTransaction`
+
+For more advanced debugging, you can directly use the `debug_traceTransaction` method.
+This method allows you to trace the execution of a transaction, providing detailed information about each call made
+during the transaction.
+
+### Manually Calling `debug_traceTransaction`
+
+1. **Set Up a Local Node**:
+    - Ensure you have a local node running that supports the `debug_traceTransaction` method.
+    - Connect to your local node using
+      a [web3 provider](https://web3js.readthedocs.io/en/v1.2.11/web3.html#currentprovider).
```

Comment:
Is this URL with tag `#currentprovider` the best place on that page to point?  I'm thinking `#setprovider` might be better?

Reply:
I think Remix lets you use `web3.currentProvider` out-of-the-box without you needing to set it manually in the script, I will check tomorrow and update this 🙌

Reply:
Added instructions on how to connect with MetaMask and also added a link to Remix's documentation on using MetaMask as an injected provider.